x-pack/plugins/maps/public/classes/sources/ems_tms_source/ems_tms_source.js

```diff
@@ -35,7 +32,7 @@ export class EMSTMSSource extends AbstractTMSSource {
       {
         id: descriptor.id,
         type: EMSTMSSource.type,
-        isAutoSelect: _.get(descriptor, 'isAutoSelect', false),
+        isAutoSelect: !!descriptor.isAutoSelect ? !!descriptor.isAutoSelect : false,
```

Comment:
Should the constructor call createDescriptor like all other constructors? That way there is only a single logic for ensuring `isAutoSelect` is set.